SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "EngineClasses/SpatialLoadBalanceEnforcer.h"
+#include "EngineClasses/SpatialVirtualWorkerTranslator.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "Interop/Connection/SpatialWorkerConnection.h"
+#include "Interop/SpatialStaticComponentView.h"
+#include "Schema/AuthorityIntent.h"
+#include "SpatialCommonTypes.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialLoadBalanceEnforcer);
+
+using namespace SpatialGDK;
+
+USpatialLoadBalanceEnforcer::USpatialLoadBalanceEnforcer(const FObjectInitializer& ObjectInitializer)
+	: Super(ObjectInitializer)
+	, NetDriver(nullptr)
+	, VirtualWorkerTranslator(nullptr)
+{
+}
+
+void USpatialLoadBalanceEnforcer::Init(USpatialNetDriver* InNetDriver, SpatialVirtualWorkerTranslator* InVirtualWorkerTranslator)
+{
+	NetDriver = InNetDriver;
+	VirtualWorkerTranslator = InVirtualWorkerTranslator;
+}
+
+void USpatialLoadBalanceEnforcer::Tick()
+{
+	ProcessQueuedAclAssignmentRequests();
+}
+
+void USpatialLoadBalanceEnforcer::OnComponentUpdated(const Worker_ComponentUpdateOp& Op)
+{
+	check(NetDriver)
+	check(NetDriver->StaticComponentView)
+	if (Op.update.component_id == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID &&
```

Comment:
Why pass any other component updates to this object? Should this just be a check?

Reply:
Done